From b3fd0abe777204781c02234fa481f3e8463f47cb Mon Sep 17 00:00:00 2001
From: Lian Chen <lian.chen@mediatek.com>
Date: Thu, 14 Jul 2022 16:33:58 +0800
Subject: [PATCH 08/19] mac80211: mtk: make 4addr null frames using min_rate
 for WDS

WDS needs 4addr packets to trigger AP for wlan0.sta creation.
However, the 4addr null frame is sent at a high rate
so that AP can't receive it.
This PATCHES switches to use min_rate.

Signed-off-by: Lian Chen <lian.chen@mediatek.com>
---
 net/mac80211/mlme.c | 1 +
 1 file changed, 1 insertion(+)

--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -1611,6 +1611,7 @@ void ieee80211_send_pspoll(struct ieee80
 	pspoll->frame_control |= cpu_to_le16(IEEE80211_FCTL_PM);
 
 	IEEE80211_SKB_CB(skb)->flags |= IEEE80211_TX_INTFL_DONT_ENCRYPT;
+	IEEE80211_SKB_CB(skb)->flags |= IEEE80211_TX_CTL_USE_MINRATE;
 	ieee80211_tx_skb(sdata, skb);
 }
 
